test/data/test_dataset.py

```diff
@@ -342,6 +342,111 @@ def test_stratified_dataset_split(self):
+        label1 = [1]
+        sentence21 = [["nay", "answer", "me"]]
+        sentence22 = [["stand", "unfold", "yourself"]]
+        label2 = [0]
+        sentence31 = [["is", "Horatio", "there"]]
+        sentence32 = [["a", "piece", "of", "him"]]
+        label3 = [0]
+
+        example1_values = sentence11 + sentence12 + label1
+        example2_values = sentence21 + sentence22 + label2
+        example3_values = sentence31 + sentence32 + label3
+
+        # Test filter remove words from single field only
+        dataset, text_field = self.filter_init(
+            example1_values,
+            example2_values,
+            example3_values
+        )
+
+        text_field.vocab.stoi.pop("there")
+        text_field.vocab.stoi.pop("bernardo")
+
+        dataset.filter_examples(["text1"])
+
+        assert dataset[0].text1 == ["who", "is"]
+        assert dataset[0].text2 == ["bernardo", "is", "there"]
+        assert dataset[0].label == 1
+
+        assert dataset[1].text1 == ["nay", "answer", "me"]
+        assert dataset[1].text2 == ["stand", "unfold", "yourself"]
+        assert dataset[1].label == 0
+
+        assert dataset[2].text1 == ["is", "Horatio"]
+        assert dataset[2].text2 == ["a", "piece", "of", "him"]
+        assert dataset[2].label == 0
+
+        # Test filter remove words from multiple fields
+        dataset, text_field = self.filter_init(
+            example1_values,
+            example2_values,
+            example3_values
+        )
+
+        text_field.vocab.stoi.pop("there")
+        text_field.vocab.stoi.pop("bernardo")
+
+        dataset.filter_examples(["text1", "text2"])
+
+        assert dataset[0].text1 == ["who", "is"]
+        assert dataset[0].text2 == ["is"]
+        assert dataset[0].label == 1
+
+        assert dataset[1].text1 == ["nay", "answer", "me"]
+        assert dataset[1].text2 == ["stand", "unfold", "yourself"]
+        assert dataset[1].label == 0
+
+        assert dataset[2].text1 == ["is", "Horatio"]
+        assert dataset[2].text2 == ["a", "piece", "of", "him"]
+        assert dataset[2].label == 0
+
+        # Test filter remove all words in example
+        dataset, text_field = self.filter_init(
+            example1_values,
+            example2_values,
+            example3_values
+        )
+
+        text_field.vocab.stoi.pop("who")
+        text_field.vocab.stoi.pop("is")
+        text_field.vocab.stoi.pop("there")
+
+        dataset.filter_examples(["text1", "text2"])
+
+        assert dataset[0].text1 == []
+        assert dataset[0].text2 == ["bernardo"]
+        assert dataset[0].label == 1
+
+        assert dataset[1].text1 == ["nay", "answer", "me"]
+        assert dataset[1].text2 == ["stand", "unfold", "yourself"]
+        assert dataset[1].label == 0
+
+        assert dataset[2].text1 == ["Horatio"]
+        assert dataset[2].text2 == ["a", "piece", "of", "him"]
+        assert dataset[2].label == 0
+
+    def filter_init(self, ex_val1, ex_val2, ex_val3):
```

Comment:
Could you move this method outside of the Test class (and add a comment that it is used in the filtering test)

Reply:
Sure